tf2/src/buffer_core.cpp

```diff
@@ -1296,11 +1354,16 @@ void BufferCore::_getFrameStrings(std::vector<std::string> & vec) const
   boost::mutex::scoped_lock lock(frame_mutex_);
 
   TransformStorage temp;
+  if (frameIDs_reverse.size() <= 1u)
+  {
+    return;
+  }
```

Comment:
The "skip NO_PARENT" part is already dealt with by the begin()+1 below. You just need the assert to document that you NO_PARENT won't go away.